### PR TITLE
W-12357357 retain empty string fields on parsing v23 duke

### PR DESCRIPTION
**Pull request by dukesphere**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc [hunk 1]
```diff
@@ -82,6 +82,7 @@ Valid values are `true` or `false`.
 Note that in Mule versions 4.0.4 and later, this is only used as a separator
 when there are multiple records. Values translate directly to character codes
 (`none` leaves no termination on each record).
+| `retainEmptyStringFieldsOnParsing` | `false` | This property handles fields that lack values. When set to `true`, the property makes the reader retain these fields and set their values to an empty string. The default behavior of the reader is to remove those fields. _Introduced in the January 2023 version of DataWeave 2.3 for Mule version 4.3.0-20221212._
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]:
```suggestion
| `retainEmptyStringFieldsOnParsing` | `false` | To handle fields that lack values, this property makes the reader retain these fields and set their values to an empty string when set to `true`. The default behavior of the reader is to remove those fields. _Introduced in the January 2023 version of DataWeave 2.3 for Mule version 4.3.0-20221212._
```

File: modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc [hunk 2]
```diff
@@ -82,6 +82,7 @@ Valid values are `true` or `false`.
 Note that in Mule versions 4.0.4 and later, this is only used as a separator
 when there are multiple records. Values translate directly to character codes
 (`none` leaves no termination on each record).
+| `retainEmptyStringFieldsOnParsing` | `false` | To handle fields that lack values, this property makes the reader retain these fields and set their values to an empty string when set to `true`. The default behavior of the reader is to remove those fields. _Introduced in the January 2023 version of DataWeave 2.3 for Mule version 4.3.0-20221212._
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]:
```suggestion
| `retainEmptyStringFieldsOnParsing` | `false` | When set to `true`, this property makes the reader retain fields that lack values and set their values to an empty string. _Introduced in the January 2023 version of DataWeave 2.3 for Mule version 4.3.0-20221212._
```

File: modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc [hunk 3]
```diff
@@ -82,6 +82,7 @@ Valid values are `true` or `false`.
 Note that in Mule versions 4.0.4 and later, this is only used as a separator
 when there are multiple records. Values translate directly to character codes
 (`none` leaves no termination on each record).
+| `retainEmptyStringFieldsOnParsing` | `false` | When set to `true`, this property makes the reader retain fields that lack values and set their values to an empty string. _Introduced in the January 2023 version of DataWeave 2.3 for Mule version 4.3.0-20221212._
```
Comment (sms-asalanova) [inline modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]: The property impacts only for String fields. May be we could clarify it.

Comment (dukesphere) [reply modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]: @sms-asalanova I need a commit suggestion please.

Comment (dukesphere) [reply modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]: That also needs to be reconciled with the approved release note. Please assist for the release.

**Review by afelisatti — APPROVED** [review]

File: modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc (hunk 3 above)
Comment (dukesphere) [reply modules/ROOT/pages/_partials/dataweave-formats-flatfile-config.adoc:85]: We've decided to improve this later if necessary. Feel free to ping me about the text you want.

**Review by dmerlob — APPROVED** [review]

**Pull request merged**